runtime/vm/ContinuationHelpers.cpp

```diff
@@ -1253,10 +1261,14 @@ takeVirtualThreadListToUnblock(J9VMThread *currentThread)
 					 *
 					 * All blocking/waiting monitors have to be inflated. If the monitor has not been inflated,
 					 * then the owner has not yet released the flat lock.
+					 *
+					 * There is a special case where the FLC bit maybe overwritten by fast exit code,
+					 * so also unblock thread if the blocking object has a unowned flatlock.
 					 */
-					if (J9_ARE_ANY_BITS_SET(((J9ThreadMonitor *)monitor)->flags, J9THREAD_MONITOR_INFLATED)
-						&& (0 == monitor->count)
-					) {
+					if (isUnownedFlatlock
+						||(J9_ARE_ANY_BITS_SET(((J9ThreadMonitor *)monitor)->flags, J9THREAD_MONITOR_INFLATED)
```

Comment:
Format here. Space after `||`. It will be easier to read if you do sth like:
```
isUnownedInflatedLock = J9_ARE_ANY_BITS_SET(((J9ThreadMonitor *)monitor)->flags, J9THREAD_MONITOR_INFLATED) && (0 == monitor->count);

if (isUnownedFlatlock || isUnownedInflatedLock)
 ) {
```